src/components/LibraryContainer.tsx

```diff
@@ -139,7 +141,48 @@ export class LibraryContainer extends React.Component<LibraryContainerProps, Lib
         this.props.libraryController.raiseEvent(name, params);
     }
 
+    // Update the selectionIndex. Current index will add by 1 if selectNextItem is true,
+    // minus by 1 otherwise, but it should always be between 0 and maxSelectionIndex.
+    updateSelectionIndex(selectNextItem: boolean) {
+        if (!this.state.inSearchMode) {
+            return;
+        }
+
+        let nextIndex = selectNextItem ? this.selectionIndex + 1 : this.selectionIndex - 1;
+
+        if (nextIndex < 0 && this.maxSelectionIndex >= 0) {
+            nextIndex = 0;
+        }
+
+        if (nextIndex >= this.maxSelectionIndex) {
+            nextIndex = this.maxSelectionIndex;
+        }
+
+        this.selectionIndex = nextIndex;
+    }
+
+    getSelectionIndex(): number {
+        return this.selectionIndex;
+    }
+
+    // New selectionIndex will be set when user clicks on an item.
+    setSelectionIndex(index: number) {
+        this.selectionIndex = index;
+        this.forceUpdate();
+    }
+
+    // Set the max selection index(number of items in the search result).
+    setMaxSelectionIndex(max: number) {
+        this.maxSelectionIndex = max;
+    }
+
     onSearchModeChanged(inSearchMode: boolean) {
+        // Reset selectionIndex when quitting from search mode
+        if (this.state.inSearchMode && !inSearchMode) {
```

Comment:
This function is called even if user just keeps typing whitespaces on the search bar. Under this case, inSearchMode is always set to false, but the index doesn't need to be updated.